pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -154,43 +169,62 @@ type mgr struct {
 }
 
 func (m *mgr) createDefaultProject(obj runtime.Object) (runtime.Object, error) {
-	return v3.ClusterConditionconditionDefautlProjectCreated.DoUntilTrue(obj, func() (runtime.Object, error) {
+	return m.createProject("Default", v3.ClusterConditionconditionDefaultProjectCreated, obj, defaultProjectLabels, defaultProjects)
+}
+
+func (m *mgr) createSystemProject(obj runtime.Object) (runtime.Object, error) {
+	return m.createProject("System", v3.ClusterConditionconditionSystemProjectCreated, obj, systemProjectLabels, systemProjects)
+}
+
+func (m *mgr) createProject(name string, cond condition.Cond, obj runtime.Object, labels labels.Set, projectMap map[string]bool) (runtime.Object, error) {
+	return cond.DoUntilTrue(obj, func() (runtime.Object, error) {
 		metaAccessor, err := meta.Accessor(obj)
 		if err != nil {
 			return obj, err
 		}
-
-		projects, err := m.projectLister.List(metaAccessor.GetName(), defaultProjectLabels.AsSelector())
+		projects, err := m.projectLister.List(metaAccessor.GetName(), labels.AsSelector())
 		if err != nil {
 			return obj, err
 		}
 		if len(projects) > 0 {
 			return obj, nil
 		}
-
+		existing := map[string]bool{}
+		var toCreate []string
+		for _, project := range projects {
+			existing[project.Spec.DisplayName] = true
```

Comment:
DisplayName is editable by the end user right? If they changed the display name, would that cause us to create more than one default and/or system project?

Reply:
generally speaking, why did you introduce this additional logic around comparing existing and toCreate?

Reply:
Projects are fetched with labels, so only projects created by us (having label `authz.management.cattle.io/default-project`) will be fetched, thus they won't get created. 

Oh, also we won't create since that happens only if the condition is true (eg `ClusterConditionconditionDefaultProjectCreated`) 

@alena1108 can confirm but I think she just wanted to check if the project already exists before creating using `toCreate` and `existing`.